SendGrid/SendGrid/Client.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Text;
+using SendGrid.Resources;
+using System.Web.Script.Serialization;
+using System.Net;
+
+namespace SendGrid
+{
+    public class Client
+    {
+        private string _apiKey;
+        public APIKeys ApiKeys;
+        private Uri _baseUri;
+
+        /// <summary>
+        ///     Create a client that connects to the SendGrid Web API
+        /// </summary>
+        /// <param name="apiKey">Your SendGrid API Key</param>
+        /// <param name="baseUri">Base SendGrid API Uri</param>
+        public Client(string apiKey, string baseUri = "https://api.sendgrid.com/")
```

Comment:
I don't remember if SendGrid apiKey's had a specific format, but it might worth checking to see if the apiKey satisfies a pre-condition format and does like an ArgumentException should the apiKey not satisfy the format.

Ex: `SG_xxxxx.xxxxx.xxx.....` (if we don't find the message is in that format)